verifier/src/advisories.rs

```diff
@@ -0,0 +1,231 @@
+/// ```
+#[derive(Debug, PartialEq, Eq, PartialOrd, Hash, Copy, Clone, Deserialize)]
+pub enum AdvisoryStatus {
+    /// TCB level of SGX platform is revoked. The platform is not trustworthy.
+    Revoked,
+    /// TCB level of SGX platform is outdated and additional configuration of
+    /// SGX platform may be needed.
+    OutOfDateConfigurationNeeded,
+    /// TCB level of SGX platform is outdated.
+    OutOfDate,
+    /// TCB level of the SGX platform is up-to-date but additional configuration
+    /// for the platform and SW Hardening in the attesting SGX enclaves may be
+    /// needed.
+    ConfigurationAndSWHardeningNeeded,
+    /// TCB level of the SGX platform is up-to-date but additional configuration
+    /// of SGX platform may be needed.
+    ConfigurationNeeded,
+    /// TCB level of the SGX platform is up-to-date but due to certain issues
+    /// affecting the platform, additional SW Hardening in the attesting SGX
+    /// enclaves may be needed.
+    SWHardeningNeeded,
+    /// TCB level of the SGX platform is up-to-date.
+    UpToDate,
+}
+
+/// The advisories pertaining to a TCB(Trusted Computing Base).
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+pub struct Advisories {
+    ids: BTreeSet<String>,
+    status: AdvisoryStatus,
+}
+
+impl Advisories {
+    /// Create a new instance.
+    ///
+    /// # Arguments:
+    /// * ids - The IDs, these are of the form "INTEL-SA-12345". They should be
+    ///   all caps.
+    /// * status - The status of the advisories specified in `ids`.
+    pub fn new<'a, I, E>(ids: I, status: AdvisoryStatus) -> Self
+    where
+        I: IntoIterator<Item = &'a E>,
+        E: ToString + 'a + ?Sized,
+    {
+        let ids = ids.into_iter().map(ToString::to_string).collect();
+        Self { ids, status }
+    }
+
+    /// Returns `true` if `self` is a superset of `other`.
+    ///
+    /// This means that `self` contains at all the advisories in `other`
+    /// and the `status` on `other` is as good or better than `status` on self.
+    fn is_superset(&self, other: &Self) -> bool {
+        if self.status > other.status {
+            return false;
+        }
+        self.ids.is_superset(&other.ids)
+    }
+}
+
+impl Display for Advisories {
+    fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        write!(f, "IDs: ")?;
+        let ids = &self.ids;
+        if ids.is_empty() {
+            write!(f, "(none)")?
+        } else {
+            write!(f, "{ids:?}")?
+        }
+        write!(f, " Status: {:?}", self.status)
+    }
+}
+
+impl SpacedStructName for Advisories {
+    fn spaced_struct_name() -> &'static str {
+        "advisories"
+    }
+}
+
+/// Verifier for ensuring the expected [`Advisories`] are the only advisories
+/// present and that they have a status that is at least as good as the
+/// expected.
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct AdvisoriesVerifier {
+    expected: Advisories,
```

Comment:
I considered naming this field "allowed" but kept it "expected" for consistency with the other Verifiers.